File: MidtermSukhadaSheth/MidTermCurrencyConverter/app/src/main/java/com/example/midtermcurrencyconverter/MainActivity.java
package com.example.midtermcurrencyconverter;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {


    Button convertButton;
    TextView fracsOutput;
    EditText dollarsInput;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.fragment_main);


        convertButton=(Button) findViewById(R.id.button);
        dollarsInput =(EditText) findViewById(R.id.editText3);
        fracsOutput = (TextView) findViewById(R.id.textView3);

        convertButton.setOnClickListener(new View.OnClickListener() {
                                                     @Override
                                                     public void onClick(View v) {
                  try {
                        Double usd = Double.parseDouble(dollarsInput.getText().toString());
                        Double result = usd * 439.36 ;
                        fracsOutput.setText("Fracs : " + Double.toString(result));
                      } catch (Exception e) {

                      } } //listener
                                                 }
        );
    }
}

File: Kotlin Codelabs/Codelabs/DiceRoller 1.2/app/src/main/java/com/example/android/diceroller/MainActivity.kt
package com.example.android.diceroller

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import java.util.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val rollButton: Button = findViewById(R.id.roll_button)
        rollButton.setOnClickListener { rollDice() }
        val countButton: Button = findViewById(R.id.count_button)
        countButton.setOnClickListener { countUp() }
        val resetButton: Button = findViewById(R.id.reset_button)
        resetButton.setOnClickListener { resetDice() }
    }
    private fun rollDice() {
        //Toast.makeText(this, "button clicked",
          //  Toast.LENGTH_SHORT).show()
        val randomInt = Random().nextInt(6) + 1
        val resultText: TextView = findViewById(R.id.result_text)
        resultText.text = randomInt.toString()
    }

    private fun resetDice() {
        val resultText: TextView = findViewById(R.id.result_text)
        resultText.text = 0.toString()
    }

    private fun countUp() {
        val resultText: TextView = findViewById(R.id.result_text)

        // If text is the default "Hello World!" set that text to 1.
        if (resultText.text == "Hello World!") {
            resultText.text = "1"
        } else {
            // Otherwise, increment the number up to 6.
            // The text value in resultText.text is an instance of the CharSequence class;
            // it needs to be converted to a String object before it can be converted to an int.
            var resultInt = resultText.text.toString().toInt()

            if (resultInt < 6) {
                resultInt++
                resultText.text = resultInt.toString()
            }
        }
    }
}
